### PR TITLE
test: Split block from citation

### PR DESCRIPTION
**Pull request by dcalhoun**
## What?

Automate testing splitting of Quote and Pullquote blocks from the citation
field.

## Why?

Improve product stability.

## How?

Update existing tests to include pressing <kbd>Return</kbd> with the cursor
positioned at the end of the citation field.

## Testing Instructions

n/a

### Testing Instructions for Keyboard

n/a

## Screenshots or screencast 
n/a

### CODE REVIEW

File: packages/block-library/src/pullquote/test/edit.native.js [hunk 1]
```diff
@@ -45,10 +46,13 @@ describe( 'Pullquote', () => {
...
-		typeInRichText( citationTextInput, 'A person', {
-			finalSelectionStart: 2,
-			finalSelectionEnd: 2,
+		typeInRichText( citationTextInput, 'A person' );
+		fireEvent( citationTextInput, 'onKeyDown', {
+			nativeEvent: {},
+			preventDefault() {},
+			keyCode: ENTER,
 		} );
+		selectRangeInRichText( citationTextInput, 2, 2 );
```
Comment (fluiddot) [inline packages/block-library/src/pullquote/test/edit.native.js:55]: Nitpick: We don't need to pass the `end` argument as it's populated with `start` value by default.
```suggestion
		selectRangeInRichText( citationTextInput, 2 );
```

https://github.com/WordPress/gutenberg/blob/5c3d93a949faff5062b158378364326064156d18/test/native/integration-test-helpers/rich-text-select-range.js#L14

File: packages/block-library/src/quote/test/edit.native.js [hunk 2]
```diff
@@ -62,10 +63,13 @@ describe( 'Quote', () => {
...
-		typeInRichText( citationTextInput, 'A person', {
-			finalSelectionStart: 2,
-			finalSelectionEnd: 2,
+		typeInRichText( citationTextInput, 'A person' );
+		fireEvent( citationTextInput, 'onKeyDown', {
+			nativeEvent: {},
+			preventDefault() {},
+			keyCode: ENTER,
 		} );
+		selectRangeInRichText( citationTextInput, 2, 2 );
```
Comment (fluiddot) [inline packages/block-library/src/quote/test/edit.native.js:72]: Nitpick: We don't need to pass the `end` argument as it's populated with `start` value by default.
```suggestion
		selectRangeInRichText( citationTextInput, 2 );
```

https://github.com/WordPress/gutenberg/blob/5c3d93a949faff5062b158378364326064156d18/test/native/integration-test-helpers/rich-text-select-range.js#L14

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 ! Thanks @dcalhoun for expanding these tests 🏅 !

File: packages/block-library/src/pullquote/test/edit.native.js (hunk 1 above)
Comment (dcalhoun) [reply packages/block-library/src/pullquote/test/edit.native.js:55]: I keep overlooking this. 😄 Resolved in https://github.com/WordPress/gutenberg/pull/53112.

File: packages/block-library/src/quote/test/edit.native.js (hunk 2 above)
Comment (dcalhoun) [reply packages/block-library/src/quote/test/edit.native.js:72]: I keep overlooking this. 😄 Resolved in https://github.com/WordPress/gutenberg/pull/53112.

**Pull request merged**